File: src/ru/edu/vtisov/array/slidingwindow/RemoveColoredPiecesIfBothNeighborsAreTheSameColor.java
package ru.edu.vtisov.array.slidingwindow;

// https://leetcode.com/problems/remove-colored-pieces-if-both-neighbors-are-the-same-color

public class RemoveColoredPiecesIfBothNeighborsAreTheSameColor {

    public boolean winnerOfGame(String colors) {
        int alice = 0;
        int bob = 0;

        for (int i = 1; i < colors.length() - 1; i++) {
            if (colors.charAt(i - 1) == colors.charAt(i) && colors.charAt(i) == colors.charAt(i + 1)) {
                if (colors.charAt(i) == 'A') {
                    alice++;
                } else {
                    bob++;
                }
            }
        }
        return alice - bob >= 1;
    }
}

File: src/ru/edu/vtisov/array/slidingwindow/remove-colored-pieces-if-both-neighbors-are-the-same-color.go
package slidingwindow

func winnerOfGame(colors string) bool {
	alice := 0
	bob := 0
	for i := 1; i < len(colors)-1; i++ {
		if colors[i-1] == colors[i] && colors[i] == colors[i+1] {
			if colors[i] == 'A' {
				alice++
			} else {
				bob++
			}
		}
	}
	return alice-bob >= 1
}
